roomserver/storage/shared/storage.go

```diff
@@ -322,14 +322,18 @@ func (d *Database) Events(
 		if roomID, ok := d.Cache.GetRoomServerRoomID(roomNID); ok {
 			roomVersion, _ = d.Cache.GetRoomVersion(roomID)
 		}
+		eventID := ""
+		if id, ierr := d.EventsTable.SelectEventID(ctx, nil, eventJSON.EventNID); ierr == nil {
```

Comment:
Can we not do this in bulk before entering the for loop? This will be kinda slow otherwise. E.g `map[EventNID]string`